tests/test_cms_menus.py

```diff
@@ -839,6 +836,91 @@ def test_menu_with_softroot_page_rendering(self):
 
         self.assertTreeQuality(soft_root, mock_tree, 'level', 'title')
 
+    def test_menu_with_hidden_softroot_page_rendering(self):
+        """
+        Tree in fixture :
+               root
+                   aaa (soft_root)
+                       aaa1
+                           ccc
+                               ddd
+                       aaa2
+                   bbb
+        tag: show_menu 0 100 100 100
+        expected result:
+            renders just the softroot page, with no menu nodes
+        """
+        menu_content_ver = factories.MenuContentWithVersionFactory(version__state=PUBLISHED, language=self.language)
+        root = factories.ChildMenuItemFactory(parent=menu_content_ver.root, content=self.root_pagecontent.page)
+        aaa = factories.ChildMenuItemFactory(
+            parent=root,
+            soft_root=True,
+            hide_node=True,
+            content=self.aaa_pagecontent.page
+        )
+        aaa1 = factories.ChildMenuItemFactory(parent=aaa, content=self.aaa1_pagecontent.page)
+        ccc = factories.ChildMenuItemFactory(parent=aaa1, content=self.ccc_pagecontent.page)
+        factories.ChildMenuItemFactory(parent=ccc, content=self.ddd_pagecontent.page)
+        factories.ChildMenuItemFactory(parent=aaa, content=self.aaa2_pagecontent.page)
+        factories.ChildMenuItemFactory(parent=root, content=self.bbb_pagecontent.page)
+
+        page = self.aaa_pagecontent.page
+        context = self.get_context(page.get_absolute_url(), page=page)
+        tpl = Template("{% load menu_tags %}{% show_menu 0 100 100 100 %}")
+        tpl.render(context)
+        soft_root = context['children']
+        mock_tree = []
+
+        self.assertTreeQuality(soft_root, mock_tree)
+
+    def test_menu_rendering_child_of_hidden_softroot_page(self):
+        """
+        Tree in fixture :
+               root
+                   aaa (soft_root)
+                       aaa1
+                           ccc
+                               ddd
+                       aaa2
+                   bbb
+        tag: show_menu 0 100 100 100
+        expected result when rendering child node ccc of hidden softroot node aaa:
```

Comment:
Why is this the expected result?

I'm guessing that it's because when hiding a parent node the child node should not render a parent node that is soft_root and hidden.